File: docs/docs/api-reference/iam/api_user/v1/examples/python/get-api-user-by-key-hash.py
from meshtrade.iam.api_user.v1 import (
    ApiUserServiceGRPCClient,
    ClientOptions,
    GetApiUserByKeyHashRequest,
)


def main():
    # Create client (loads credentials from MESH_API_CREDENTIALS)
    options = ClientOptions()
    client = ApiUserServiceGRPCClient(options)
    
    with client:
        # Create request
        request = GetApiUserByKeyHashRequest(
            key_hash="abcd1234hash5678"  # Replace with actual key hash
        )
        
        # Make the API call
        response = client.get_api_user_by_key_hash(request)
        
        print(f"Found API User: {response.name}")
        print(f"Display Name: {response.display_name}")
        print(f"State: {response.state}")


if __name__ == "__main__":
    main()

File: docs/docs/api-reference/iam/api_user/v1/examples/go/get-api-user-by-key-hash.go
package main

import (
	"context"
	"log"

	api_userv1 "github.com/meshtrade/api/go/iam/api_user/v1"
)

func main() {
	// Create client (see client-setup.go for details)
	client, err := api_userv1.NewApiUserServiceGRPCClient(
		api_userv1.WithAddress("localhost", 8080),
		api_userv1.WithTLS(false),
		api_userv1.WithAPIKey("your-api-key"),
		api_userv1.WithGroup("your-group-id"),
	)
	if err != nil {
		log.Fatalf("Failed to create client: %v", err)
	}
	defer client.Close()

	// Get API user by key hash (typically used for authentication flows)
	apiUser, err := client.GetApiUserByKeyHash(
		context.Background(),
		&api_userv1.GetApiUserByKeyHashRequest{
			KeyHash: "abcd1234hash5678", // Replace with actual key hash
		},
	)
	if err != nil {
		log.Fatalf("Failed to get API user by key hash: %v", err)
	}

	log.Printf("Found API User: %s", apiUser.Name)
	log.Printf("Display Name: %s", apiUser.DisplayName)
	log.Printf("State: %v", apiUser.State)
}
